### PR TITLE
Add support for Windows

### LINKED ISSUE #5: Support for windows
It seems like we could get the default browser on Windows by running a command like

```
reg QUERY HKEY_CURRENT_USER\Software\Microsoft\Windows\Shell\Associations\UrlAssociations\http\UserChoice /v ProgId
```

Which prints output like

```
HKEY_CURRENT_USER\Software\Microsoft\Windows\Shell\Associations\UrlAssociations\http\UserChoice
    ProgId    REG_SZ    ChromeHTML
```

Or

```
HKEY_CURRENT_USER\Software\Microsoft\Windows\Shell\Associations\UrlAssociations\http\UserChoice
    ProgId    REG_SZ    MSEdgeDHTML
```

We wouldn't be able to get a similarly compatible `id`, as far as I can see, but we could categorize and retrieve the `name`. E.g: https://github.com/microsoft/vscode-js-debug/blob/master/src/common/defaultBrowserProvider.ts

Would you be open to a PR to add support for this?

### PR DESCRIPTION
**Pull request by connor4312**
I had to adjust some linting rules as it looks like `xo` has changed in the years since this module was first written. I've also kept with the `>=4` version constraint, which makes code less pretty. I can update that to >=8 as part of this PR, or a separate PR--let me know if you want that 🙂 

Fixes #5

### CODE REVIEW

**Comment by sindresorhus** [comment]
> I can update that to >=8 as part of this PR

Yeah, feel free to require Node.js 10 in this PR, and also update the dependencies.

File: index.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 const defaultBrowserId = require('default-browser-id');
 const bundleName = require('bundle-name');
+const windows = require('./windows');
```
Comment (sindresorhus) [inline index.js:4]: You need to add this file to `files` in package.json.

**Comment by shazron** [comment]
@connor4312 will you have time to update the PR? if not I can use your code and submit a new PR based on it, with the changes.

**Comment by connor4312** [comment]
I've updated the PR :)

**Comment by connor4312** [comment]
It looks like `com.apple.launchservices.secure.plist` isn't available on the OSX machines that Travis uses: https://travis-ci.com/github/sindresorhus/default-browser/jobs/382419730

**Comment by sindresorhus** [comment]
> Yeah, feel free to require Node.js 10 in this PR, and also update the dependencies.

Node.js **10**

File: package.json [hunk 2]
```diff
@@ -37,13 +38,22 @@
   "dependencies": {
     "bundle-name": "^2.0.1",
     "default-browser-id": "^2.0.0",
-    "xdg-default-browser": "^2.0.0"
+    "execa": "^0.10.0",
+    "xdg-default-browser": "^2.1.0"
```
Comment (sindresorhus) [inline package.json:42]: Upgrade the dependencies to latest.

File: windows.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+const execa = require('execa');
+const util = require('util');
+
+// Windows doesn't have browser IDs in the same way OSX/Linux does--give fake
```
Comment (sindresorhus) [inline windows.js:5]: `OSX` => `macOS`

File: windows.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+// ones that look real and match the OSX/Linux versions for cross-platform apps.
+const windowsBrowserProgIds = {
+	AppXq0fevzme2pys62n3e0fbqa7peapykr8v: {name: 'Edge', id: 'com.microsoft.edge.old'},
+	MSEdgeDHTML: {name: 'Edge', id: 'com.microsoft.edge'}, // On OSX, its "com.microsoft.edgemac"
```
Comment (sindresorhus) [inline windows.js:9]: `it's`

File: windows.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	ChromeHTML: {name: 'Chrome', id: 'com.google.chrome'}
+};
+
+function UnknownBrowserError(message) {
+	Error.captureStackTrace(this, UnknownBrowserError);
+	this.name = this.constructor.name;
+	this.message = message;
+}
+
+util.inherits(UnknownBrowserError, Error);
```
Comment (sindresorhus) [inline windows.js:21]: Use normal ES6 subclassing.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
You need to mention Windows support in the readme.

**Comment by DannyFeliz** [comment]
@connor4312 👀

**Comment by connor4312** [comment]
Thanks for the ping. Updated dependencies as requested. The update to tsd caused an issue in tests complaining about a lack of `esModuleInterop` for the default import, so I changed that to an `import *`.

**Comment by connor4312** [comment]
Looks like the test failure is caused by `default-browser-id`. Not sure what the deal there is -- the module works on Catalina and Big Sur, maybe Github Actions' macOS setup is set up differently than a standard osx install.

File: package.json [hunk 6]
```diff
@@ -7,16 +7,17 @@
   "author": {
     "name": "Sindre Sorhus",
     "email": "sindresorhus@gmail.com",
-    "url": "sindresorhus.com"
+    "url": "https://sindresorhus.com"
   },
   "engines": {
-    "node": ">=4"
+    "node": ">=8"
```
Comment (sindresorhus) [inline package.json:13]: You can target Node.js 12 now.

File: windows.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	const result = await _execa('reg', [
+		'QUERY',
+		' HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice',
+		'/v',
+		'ProgId'
+	]);
+
+	// Execa@0.10 throws if the command fails--no need to check `result.failed`
+
+	const match = /ProgId\s*REG_SZ\s*(\S+)/.exec(result.stdout);
```
Comment (sindresorhus) [inline windows.js:27]: Use named capture group.

File: windows.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+module.exports = async (_execa = execa) => {
+	const result = await _execa('reg', [
+		'QUERY',
+		' HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice',
+		'/v',
+		'ProgId'
+	]);
+
+	// Execa@0.10 throws if the command fails--no need to check `result.failed`
```
Comment (sindresorhus) [inline windows.js:25]: This comment mentions an old Execa version.

File: windows.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+	const match = /ProgId\s*REG_SZ\s*(\S+)/.exec(result.stdout);
+	if (!match) {
+		throw new UnknownBrowserError(`Cannot find windows browser in stdout: ${JSON.stringify(result.stdout)}`);
```
Comment (sindresorhus) [inline windows.js:29]:
```suggestion
		throw new UnknownBrowserError(`Cannot find Windows browser in stdout: ${JSON.stringify(result.stdout)}`);
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Pull request merged**